examples/nlp/multimodal_entailment.py

```diff
@@ -0,0 +1,577 @@
+"""
+Title: Learning Multimodal Entailment
+Author: [Sayak Paul](https://twitter.com/RisingSayak)
+Date created: 2021/08/08
+Last modified: 2021/08/08
+Description: Training a multimodal model for predicting entailment.
+"""
+"""
+## Introduction
+
+In today's world information is mostly found in unstructured or semi-structured forms
+like videos, texts, images, audios, etc. Therefore, to capture their inter-relationships
+we need to allow deep models to take multimodal inputs i.e. text, images, etc. On social
+media platforms, to audit and moderate content we often need to find answers to the
+following questions in near real-time:
+
+* Does a given piece of information contradict the other?
+* Does a given piece of information imply the other?
+
+In NLP, this task is referred to as recognizing textual entailment that only deals with
+texts as inputs. But in this case, we can have different inputs modalities. In this
+example, we will build and train a model for predicting multimodal entailment. We will be
+using the
+[multimodal entailment dataset](https://github.com/google-research-datasets/recognizing-multimodal-entailment)
+recently introduced by Google Research.
+
+This example requires TensorFlow 2.5 or higher. In addition, TensorFlow Hub and
+TensorFlow Text are required for the BERT model
+([Devlin et al.](https://arxiv.org/abs/1810.04805)). These libraries can be installed
+using the following command:
+"""
+
+"""shell
+!pip install -q tensorflow_text
+"""
+
+"""
+## Collect the dataset
+
+The original dataset is available
+[here](https://github.com/google-research-datasets/recognizing-multimodal-entailment).
+However, we will be using a better prepared version of the dataset. Thanks to
```

Comment:
"we will be using a better prepared version" -> please detail how it was modified from the original and why.